client_proposal.go

```diff
@@ -0,0 +1,100 @@
+	}
+
+	// proposalHandler implements a client.ProposalHandler wrapping a prnm
+	// ProposalHandler
+	proposalHandler struct {
+		c *Client         // back-reference for ProposalResponder
+		h ProposalHandler // wrapped ProposalHandler
+	}
+
+	// A ChannelProposal describes a proposal to open a new channel.
+	//
+	// The proposer has index 0 and proposee index 1.
+	ChannelProposal struct {
+		PeerPerunID       *Address // The peer proposing the channel.
+		ChallengeDuration int64    // Proposed challenge duration in case of disputes.
+		InitBals          *BigInts // Initial channel balances.
+	}
+
+	// A ProposalResponder lets the user respond to a channel proposal. If the
+	// user wants to accept the proposal, they should call Accept(), otherwise
+	// Reject(). Only a single function must be called and every further call
+	// causes a panic.
+	ProposalResponder struct {
+		c *Client                   // back-reference for account generation in Accept
+		r *client.ProposalResponder // wrapped ProposalResponder
+	}
+)
+
+// Handle implements the client.ProposalHandler interface by converting the
+// passed types from the go-perun/client package into their local conterparts
+// and then calling the prnm.ProposalHandler.
+func (h *proposalHandler) Handle(_prop *client.ChannelProposal, _resp *client.ProposalResponder) {
+	// Security Note: we don't check the remote nonce or channel participant. If
+	// this code were to evolve to production grade, this needs to be taken care
+	// of. In this case, at least the Nonce should be part of the ChannelProposal
+	// struct, as is the case for the client.ChannelProposal.
+	prop := &ChannelProposal{
+		PeerPerunID:       &Address{*(_prop.PeerAddrs[0]).(*ethwallet.Address)},
+		ChallengeDuration: int64(_prop.ChallengeDuration),
+		InitBals:          &BigInts{_prop.InitBals.Balances[0]},
+	}
+	resp := &ProposalResponder{c: h.c, r: _resp}
+	h.h.Handle(prop, resp)
+}
+
+// Accept lets the user signal that they want to accept the channel proposal.
+// Returns the newly created channel controller if the channel was successfully
+// created and funded. Panics if the proposal was already accepted or rejected.
+//
+// After the channel got successfully created, the user is required to start the
+// update handler with PaymentChannel.HandleUpdates(UpdateHandler) and to start
+// the channel watcher with PaymentChannel.Watch() on the returned channel
+// controller.
+//
+// It is important that the passed context does not cancel before twice the
+// ChallengeDuration has passed (at least for real blockchain backends with wall
+// time), or the channel cannot be settled if a peer times out funding.
+func (r *ProposalResponder) Accept(ctx *Context) (*PaymentChannel, error) {
+	ch, err := r.r.Accept(ctx.ctx, client.ProposalAcc{
+		// Generate new account as channel participant.
+		Participant: r.c.w.NewAccount().Address(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &PaymentChannel{ch}, nil
```

Comment:
Yeah also thought about it but then wanted to do it clean and not return something that might give a false sense of a non-nil object. But you're right, since we use it in generated bindings, it doesn't matter because an exception will be thrown.